dpar/src/features/input_layers.rs

```diff
@@ -227,6 +271,89 @@ impl InputVectorizer {
             }
         }
     }
+
+    /// Vectorize a parser state into the given association measure slices.
+    ///
+    /// Add to `non_lookup_slices` the association measure between all parser state addresses
+    /// undergoing attachment.
+    pub fn realize_into_assoc_strengths(
+        &self,
+        state: &ParserState,
+        non_lookup_slices: &mut [f32],
+        attachment_addrs: &[AttachmentAddr],
+    ) {
+        for (addr, non_lookup_slice) in attachment_addrs.iter().zip(non_lookup_slices.iter_mut()) {
```

Comment:
These are zipped, however, as far as I can see, they are not the same length? `attachment_addrs` has length 2, `non_lookup_slices` has the number of dependency relations times 2. You probably want to use a `repeat` iterator for `attachment_addrs`.

Reply:
Yes, you're right, the lengths are different. I'll adjust this.